src/NServiceBus.Core/Transports/Msmq/MsmqMessageDispatcher.cs

```diff
@@ -129,12 +129,27 @@ bool IsCombiningTimeToBeReceivedWithTransactions(ReadOnlyContextBag context, Dis
                 return false;
             }
 
+            if (Transaction.Current != null)
```

Comment:
Should we explicitly put `System.Transactions.Transaction` into `TransportTransaction` when in DTC mode? That is what SQL transport does here https://github.com/Particular/NServiceBus.SqlServer/blob/develop/src/NServiceBus.SqlServer/Receiving/ReceiveWithTransactionScope.cs#L46.

It makes sense from point of view of the persistence to be able to explicitly tell if the ambient transaction has actually been used by the transport or it was created later in the pipeline.